cluster/cluster.go

```diff
@@ -79,20 +79,31 @@ func Start() {
 	log.Info("CLU Start: joined to %d nodes in cluster", n)
 }
 
+type partitionCandidates struct {
+	priority int
+	nodes    []Node
+}
+
 // return the list of nodes to broadcast requests to
-// Only 1 member per partition is returned. This list includes
-// ThisNode if it is capable of handling queries.
+// Only 1 member per partition is returned.
+// The nodes are selected based on priority, using a random selection
+// if there are multi nodes with the highest priority.
 func MembersForQuery() []Node {
 	thisNode := Manager.ThisNode()
 	// If we are running in single mode, just return thisNode
 	if Mode == ModeSingle {
 		return []Node{thisNode}
 	}
 
-	membersMap := make(map[int32][]Node)
+	// store the available nodes for each partition, grouped by
+	// priority
+	membersMap := make(map[int32]*partitionCandidates)
 	if thisNode.IsReady() {
```

Comment:
shouldn't we set the priority irrespective of `thisNode.IsReady()` ? same for a bit further down where we continue the for loop `if !member.IsReady()`.  I thought we would set the candidacies of the nodes based solely on their priority. now they are all still subject to the warmup period and we won't query them until they are warmed up.